src/harvest.cpp

```diff
@@ -0,0 +1,112 @@
+#include <algorithm>
+#include <string>
+#include "assign.h"
+#include "debug.h"
+#include "harvest.h"
+#include "item.h"
+#include "output.h"
+
+template <>
+const harvest_id string_id<harvest_list>::NULL_ID( "null" );
+
+static std::map<harvest_id, harvest_list> harvest_all;
```

Comment:
Why not an instance of `generic_factory`? It would spare you some of the boilerplate.